### PR TITLE
Add consumer system v2 sub results

### PR DESCRIPTION
**Pull request by mortendaehli**
Adding detailed results to the new Consumer System V2

### CODE REVIEW

File: src/ecalc/libraries/libecalc/common/libecalc/core/consumers/compressor/component.py [hunk 1]
```diff
@@ -94,10 +98,53 @@ def evaluate(
                 values=aggregated_result.is_valid, timesteps=evaluated_timesteps, unit=Unit.NONE
             ),
             id=self.id,
+            recirculation_loss=TimeSeriesRate(
+                values=aggregated_result.recirculation_loss,
```
Comment (jorgenengelsen) [inline src/ecalc/libraries/libecalc/common/libecalc/core/consumers/compressor/component.py:102]: What about regularity here? Should it not be included in all timeseries?

File: src/ecalc/libraries/libecalc/common/libecalc/core/consumers/consumer_system.py [hunk 2]
```diff
@@ -211,9 +216,40 @@ def evaluate(
...
+
+            sub_component_results.extend(
+                [
+                    consumer.evaluate(adjusted_operational_setting)
+                    for consumer, adjusted_operational_setting in zip(sorted_consumers, adjusted_operational_settings)
+                ]
+            )
+
+        # Change to 1-index operational_settings_used
+        for idx, value in enumerate(operational_settings_used.values):
```
Comment (jorgenengelsen) [inline src/ecalc/libraries/libecalc/common/libecalc/core/consumers/consumer_system.py:251]: idx?

**Review by jorgenengelsen — APPROVED** [review]

File: src/ecalc/libraries/libecalc/common/libecalc/core/consumers/compressor/component.py (hunk 1 above)
Comment (mortendaehli) [reply src/ecalc/libraries/libecalc/common/libecalc/core/consumers/compressor/component.py:102]: Updated :)

Not sure exactly where we want to isolate regularity. I think it is Okay to pass it down to the consumer, but we want to investigate if we can remove it completely from libecalc.core

File: src/ecalc/libraries/libecalc/common/libecalc/core/consumers/consumer_system.py (hunk 2 above)
Comment (mortendaehli) [reply src/ecalc/libraries/libecalc/common/libecalc/core/consumers/consumer_system.py:251]: Renamed to index. I think idx is a common short for index.

File: src/ecalc/libraries/libecalc/common/libecalc/core/consumers/pump/component.py [hunk 3]
```diff
@@ -81,6 +88,23 @@ def evaluate(
...
+            ),
+            inlet_pressure_bar=TimeSeriesFloat(
+                values=aggregated_result.suction_pressure,
+                timesteps=evaluated_timesteps,
+                unit=Unit.BARA,
+            ),
+            outlet_pressure_bar=TimeSeriesFloat(
+                values=aggregated_result.discharge_pressure,
+                timesteps=evaluated_timesteps,
+                unit=Unit.BARA,
```
Comment (jorgenengelsen) [inline src/ecalc/libraries/libecalc/common/libecalc/core/consumers/pump/component.py:107]: regularity may be needed here aswell then?

**Review by jorgenengelsen — CHANGES_REQUESTED** [review]

**Review by jorgenengelsen — APPROVED** [review]

**Pull request merged**